include/picongpu/particles/collision/Collider.hpp

```diff
@@ -37,21 +37,42 @@ namespace picongpu
     {
         namespace collision
         {
-            template<typename T_Collider>
+            namespace detail
+            {
+                //! For each implementation for calling a collider for each species pair with a loop index
+                template<typename T_SpeciesPairList, typename T_Collider, uint32_t colliderId>
+                struct CallColliderForAPair
+                {
+                    template<size_t... I>
+                    HINLINE void operator()(
+                        std::index_sequence<I...>,
+                        std::shared_ptr<DeviceHeap> const& deviceHeap,
+                        uint32_t currentStep)
+                    {
+                        (collision::WithPeer<
+                             ApplyGuard<typename T_Collider::Functor>,
+                             typename bmpl::at_c<T_SpeciesPairList, I>::type::first_type,
```

Comment:
in `Collider.def` `boost::mpl::pair` is replaced by `std::pair`. I am not sure if `boost::mpl::at_c` can/should still be used here.